mirrord/layer/src/tcp.rs

```diff
@@ -82,13 +85,41 @@ pub(crate) trait TcpHandler {
         }
     }
 
+    /// Handle a potential [`LayerError::NewConnectionAfterSocketClose`] error.
+    fn check_connection_handling_result(res: Result<(), LayerError>) -> Result<(), LayerError> {
+        if let Err(LayerError::NewConnectionAfterSocketClose(connection_id)) = res {
+            // This can happen and is valid:
+            // 1. User app closes socket.
+            // 2. mirrord sends `PortUnsubscribe` to agent.
+            // 3. Agent sniffs new incoming connection and sends it to the layer.
+            // 4. Agent receives `PortUnsubscribe`.
+            // Step 2 could even happen after 4.
+            if Self::IS_STEAL {
+                warn!(
```

Comment:
I'm not quite sure what our `warn` policy is, should we do it even though this is valid and returns `Ok`?

Does it impact the user?

wdyt @aviramha

Reply:
I shouldn't have written this situation is _valid_ (removed this wording in https://github.com/metalbear-co/mirrord/pull/1557/commits/a076294bd0de3fef497ebe94015ae866a9d31523).

In some circumstances it might be a hiccup for the user - without mirrord, that new connection would have been routed to the pod, and accepted if the remote app is still listening. With mirrord - that connection is not handled by either process (local or remote), and will be reset once the agent gets the `PortUnsubscribe` message, even if a new connection a second later is accepted by the remote app.

With a lot of effort we could partially handle this situation: we could introduce a new message to the protocol that tells the agent, that this happens and also messages for sending back already forwarded bytes, then the agent could try to start a connection with the remote app and be a proxy. I think it's not worth it, but I think we should give the user some indication that we swallowed that connection.

Reply:
Alright, I think this behavior is weird enough that I would also vote for a `warn` there.